### PR TITLE
quotas: refactor/expand tests

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by dcramer** [comment]
was hoping to flush out a bug here, no success

File: tests/integration/tests.py [hunk 1]
```diff
@@ -398,6 +398,25 @@ def test_protocol_v6(self):
...
+        results = []
+        for n in range(10):
+            results.append(self._postWithHeader(
+                data={'message': 'hello'},
+                key=self.projectkey.public_key,
+                secret=self.projectkey.secret_key,
+                protocol='6',
+            ))
+
+        # 1 per hour
```
Comment (tkaemming) [inline tests/integration/tests.py]: Talked about this Friday, but this needs be the first thing to happen in this test

File: src/sentry/quotas/base.py [hunk 2]
```diff
@@ -39,6 +39,33 @@ def __init__(self, **kwargs):
...
+        self.limit = limit
+        # time in seconds that this quota reflects
+        self.window = window
+        # a machine readable string
+        self.reason_code = reason_code
+        # should this quota be hard-enforced (or just tracked)
+        self.enforce = enforce
+
+    def __eq__(self, other):
+        return isinstance(other, BasicQuota) and hash(self) == hash(other)
```
Comment (tkaemming) [inline src/sentry/quotas/base.py]: Technically should match the attributes here and not the hashes

File: src/sentry/quotas/base.py [hunk 3]
```diff
@@ -47,12 +74,59 @@ class Quota(Service):
...
     __all__ = (
         'get_maximum_quota', 'get_organization_quota', 'get_project_quota',
-        'is_rate_limited', 'translate_quota', 'validate',
+        'get_quotas', 'is_rate_limited', 'translate_quota', 'validate',
     )
 
     def __init__(self, **options):
         pass
 
+    def get_actionable_quotas(self, project, key=None):
```
Comment (tkaemming) [inline src/sentry/quotas/base.py]: 👍

**Pull request closed without merging**